### PR TITLE
Align Makefile and fix golangci errors

### PR DESCRIPTION
**Pull request by stuggi**
so this is just a PR to align the Makefile and how functional tests run in CI with the other operators and fix things to make golangci happy.

Depends-On: https://github.com/openshift/release/pull/54445

### CODE REVIEW

File: api/v1beta1/galera_webhook.go [hunk 1]
```diff
@@ -121,6 +123,11 @@ func (spec *GaleraSpecCore) ValidateCreate(basePath *field.Path) (admission.Warn
 func (r *Galera) ValidateUpdate(old runtime.Object) (admission.Warnings, error) {
 	galeralog.Info("validate update", "name", r.Name)
 
+	oldGalera, ok := old.(*Galera)
```
Comment (stuggi) [inline api/v1beta1/galera_webhook.go:126]: to not set  `_` for old I added this.

File: api/test/helpers/harnesses.go [hunk 2]
```diff
@@ -62,7 +62,7 @@ type MariaDBTestHarness struct {
 	mariaDBHelper      *TestHelper
 	timeout            time.Duration
 	interval           time.Duration
-	runUpdate          bool
+	//runUpdate          bool - unused
```
Comment (abays) [inline api/test/helpers/harnesses.go:65]: Should we just remove this line entirely if it's not used?

Comment (stuggi) [reply api/test/helpers/harnesses.go:65]: I left it as I wanted to check with @zzzeek

Comment (zzzeek) [reply api/test/helpers/harnesses.go:65]: I would assume the additonal linter steps added here found that this was already unused, is that correct?   if it's not used then it's fine

**Review by zzzeek — APPROVED** [review]

File: api/test/helpers/harnesses.go (hunk 2 above)
Comment (stuggi) [reply api/test/helpers/harnesses.go:65]: yes it was reported to not be used. I just left it if you are planning for additional functionality which will need it. I'll push an update to remove it

Comment (zzzeek) [reply api/test/helpers/harnesses.go]: if i wanted it back I'd add it again :)   no idea what it was for

File: api/test/helpers/harnesses.go [hunk 3]
```diff
@@ -62,7 +62,6 @@ type MariaDBTestHarness struct {
 	mariaDBHelper      *TestHelper
 	timeout            time.Duration
 	interval           time.Duration
-	runUpdate          bool
```
Comment (zzzeek) [inline api/test/helpers/harnesses.go:65]: cc @mrkisaolamb this was committed in 0c306d23 was this a merge artifact of some kind?

**Review by zzzeek — APPROVED** [review]
if you did a grep on everything else then this should be fine, if some other controller has this flag in their test suite and then won't build, it seems to do nothing, so we'd just bump that controiller to not use it

File: api/test/helpers/harnesses.go (hunk 3 above)
Comment (mrkisaolamb) [reply api/test/helpers/harnesses.go:65]: yes this is merge artifact

**Review by mrkisaolamb — APPROVED** [review]

**Pull request merged**